core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/AbstractBox.java

```diff
@@ -72,7 +72,7 @@
   private final Collection<ChangeLog> changeLog = new ArrayList<>();
 
   @Override
-  public long getId() {
+  public Long getId() {
```

Comment:
I get changing all the `UNSAVED_ID` constants to primitive, but why change getters to `Long` object when the field is still a primitive? It just necessitates a lot of boxing/unboxing

Reply:
Because I'd like to be able to use the `Aliasable` for all the things and all the new classes we created break with `long`.

Reply:
I don't understand why Aliasable was changed since it returned primitive long until recently. Have new objects been added that have `Long` IDs? If so, they are the problem, and should conform to the same pattern as everything else

Reply:
I made `Institute` and friends extend `Aliasable`. They have `Long` ids.

Reply:
If we're keeping the IDs inconsistent, please make a ticket to fix that. IDs should either be `Long` with null checks on `getId()` to check if unsaved, or `long` with `getId() == x.UNSAVED_ID` to check if unsaved. Current state will lead to weird errors due to null checks on things that can't be null, but seem like they can.